engine/class_modules/sc_shaman.cpp

```diff
@@ -5383,6 +5387,39 @@ struct primordial_wave_t : public shaman_spell_t
   }
 };
 
+// ==========================================================================
+// Chain Harvest - Venthyr Covenant
+// ==========================================================================
+struct chain_harvest_t : public chained_base_t
+{
+  chain_harvest_t( shaman_t* player, const std::string& options_str )
+    : chained_base_t( player, "chain_harvest", player->covenant.venthyr, 0, options_str )
+  {
+    if ( !player->covenant.venthyr->ok() )
+      return;
+
+    aoe = 5;
+    spell_power_mod.direct = player->find_spell( 320752 )->effectN( 1 ).sp_coeff();
+  }
+
+  void impact( action_state_t* s ) override
+  {
+    chained_base_t::impact( s );
+
+    if ( result_is_hit( s->result ) && s->result == RESULT_CRIT )
+    {
+      // This value appears to hardcoded in the spell
+      p()->cooldown.chain_harvest->adjust( -timespan_t::from_seconds(5) );
+    }
```

Comment:
I'm not sure this is behaving as expected. As best I can tell in the output, the execution interval remains approx 90 seconds when I would expect it to go down a bit from there.

I'm wondering if the spell isn't considered to be on cool down when impact() executes, though I would expect it is based on in-game behavior.

Looking at similar effects, like Rogue's Sepsis, setting up the `cooldown_t*` and calling `adjust( timespan_t* )` on it appears to be the preferred way.